File: frontend/src/api/history.ts
import apiClient from './client';
import type { ApiResponse } from '../types/common';

export interface HistoryItem {
  id: number;
  shop_id: number;
  shop_name: string;
  shop_cover: string | null;
  region: string | null;
  viewed_at: string;
}

/** 获取当前用户的浏览历史 */
export async function fetchHistory(page = 1, pageSize = 20): Promise<{ items: HistoryItem[]; total: number; has_more: boolean }> {
  const res = await apiClient.get<ApiResponse<{ items: HistoryItem[]; total: number; has_more: boolean }>>('/users/me/history', {
    params: { page, page_size: pageSize },
  });
  return res.data.data;
}

/** 删除单条浏览历史 */
export async function deleteHistory(historyId: number): Promise<void> {
  await apiClient.delete(`/users/me/history/${historyId}`);
}

/** 清空所有浏览历史 */
export async function clearHistory(): Promise<number> {
  const res = await apiClient.delete<ApiResponse<{ deleted_count: number }>>('/users/me/history/clear');
  return res.data.data.deleted_count;
}

File: backend/routers/history.py
from fastapi import APIRouter, Depends, Query

from schemas.common import ResponseModel
from schemas.users import UserResponse
from services import LogService
from services.history_service import HistoryService
from utils.auth import require_login

router = APIRouter(tags=["浏览历史模块"])


@router.get("/users/me/history", response_model=ResponseModel, summary="浏览历史")
async def get_my_history(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    current_user: UserResponse = Depends(require_login),
):
    """
    浏览历史接口
    
    查询参数:
    - page (integer, 可选): 页码，默认值为 1
    - page_size (integer, 可选): 每页大小，默认值为 20，最大值为 100
    
    响应:
    - 成功返回浏览历史记录
    - 失败返回错误信息
    """
    data = await HistoryService.get_view_history(current_user.id, page=page, page_size=page_size)
    return ResponseModel.success(data=data)


@router.delete("/users/me/history/{history_id}", response_model=ResponseModel, summary="删除单条浏览历史")
async def delete_history(
    history_id: int,
    current_user: UserResponse = Depends(require_login),
):
    """
    删除单条浏览历史接口
    
    路径参数:
    - history_id (integer, 必填): 浏览历史ID
    
    响应:
    - 成功返回删除成功信息
    - 失败返回错误信息
    """
    ok = await HistoryService.delete(history_id)
    if ok:
        await LogService.log(action="delete_history", operator=current_user, target_type="history", target_id=history_id)
    return ResponseModel.success(data={}, message="删除成功" if ok else "记录不存在")


@router.delete("/users/me/history/clear", response_model=ResponseModel, summary="清空浏览历史")
async def clear_history(
    current_user: UserResponse = Depends(require_login),
):
    """
    清空浏览历史接口
    
    响应:
    - 成功返回删除数量
    - 失败返回错误信息
    """
    count = await HistoryService.clear(current_user.id)
    if count:
        await LogService.log(action="clear_history", operator=current_user, target_type="history", target_id=0, detail={"count": count})
    return ResponseModel.success(data={"deleted_count": count}, message=f"已清空 {count} 条记录")
